### PR TITLE
4942 attached spreadsheets styling

### PR DESCRIPTION
**Pull request by emitche**
- Add styling to linked documents

### CODE REVIEW

**Comment by emitche** [comment]
My merge wiped some of my changes and there are some tweaks I need to take care of.

File: app/assets/javascripts/backbone/views/loan_questionnaires_view.coffee [hunk 1]
```diff
@@ -17,6 +18,13 @@ class MS.Views.LoanQuestionnairesView extends Backbone.View
       @initializeTree()
       @filterSwitchView.filterInit()
 
+  removeLinkedDocument: (e) ->
+    e.preventDefault()
+    item = e.currentTarget
+    container = @$(item).closest('.linked-document')
+    @$(container).find('input[type="text"]').val('')
```
Comment (linesarefuzzy) [inline app/assets/javascripts/backbone/views/loan_questionnaires_view.coffee]: `container` should already be a jQuery object, so you shouldn't need the `@$()` here.

File: app/assets/javascripts/backbone/views/loan_questionnaires_view.coffee [hunk 2]
```diff
@@ -17,6 +18,13 @@ class MS.Views.LoanQuestionnairesView extends Backbone.View
...
+    @$(item).addClass('hidden')
```
Comment (linesarefuzzy) [inline app/assets/javascripts/backbone/views/loan_questionnaires_view.coffee]: Why not `.hide()`?

File: config/locales/en/loan_questions.en.yml [hunk 3]
```diff
@@ -17,6 +17,18 @@ en:
     required_loan_types: "Required for which loan types?"
     text: "Title"
 
+  linked_document:
+    cell_end: End Cell
+    cell_end_hint: Z26
+    cell_start: Start Cell
+    cell_start_hint: A1
+    delete: Delete Document
```
Comment (emitche) [inline config/locales/en/loan_questions.en.yml]: @rgalusca , this is the line you would need to change. The text is only used once as the text for the delete link.

File: app/assets/javascripts/backbone/views/loan_questionnaires_view.coffee (hunk 1 above)
Comment (emitche) [inline app/assets/javascripts/backbone/views/loan_questionnaires_view.coffee]: I will fix.

File: app/assets/javascripts/backbone/views/loan_questionnaires_view.coffee (hunk 2 above)
Comment (emitche) [inline app/assets/javascripts/backbone/views/loan_questionnaires_view.coffee]: I have no preference. Will try what you have suggested.

File: app/assets/stylesheets/admin/_loan_questions.scss [hunk 4]
```diff
@@ -11,3 +11,42 @@
...
+
+  .ld-container {
+    margin-left: 1em;
+  }
+
+  .form-group {
+    margin: .5em 2em .25em 0;
+  }
+
+  @media (min-width: 768px) {
```
Comment (linesarefuzzy) [inline app/assets/stylesheets/admin/_loan_questions.scss]: Thanks for checking styles on mobile.

File: app/views/admin/loans/content/_questionnaire_group.html.slim [hunk 5]
```diff
@@ -98,31 +98,52 @@
...
+                      class: 'form-control', placeholder: t("linked_document.cell_start_hint")
+                    .form-group
+                      label for="#{field.attribute_sym}__end_cell" = t("linked_document.cell_end")
+                      = f.input_field "#{field.attribute_sym}__end_cell".to_sym, as: :string,
+                      class: 'form-control', placeholder: t("linked_document.cell_end_hint")
+                    - if document
+                      - if document.display_url.present?
+                        .doc-actions
+                          a [data-action="delete"
+                            data-confirm== t("linked_document.delete_confirmation")
```
Comment (linesarefuzzy) [inline app/views/admin/loans/content/_questionnaire_group.html.slim]: These double equals aren't necessary. Slim already interprets anything directly after the equals as a ruby expression.

File: app/views/admin/loans/content/_questionnaire_group.html.slim [hunk 6]
```diff
@@ -98,31 +98,52 @@
...
+                            data-confirm-cancel== t(:cancel)
+                            data-confirm-proceed== t(:confirm)
```
Comment (linesarefuzzy) [inline app/views/admin/loans/content/_questionnaire_group.html.slim]: Aren't these generic button names set as defaults in the js confirm config file?

File: app/views/admin/loans/content/_questionnaire_group.html.slim [hunk 7]
```diff
@@ -98,31 +98,52 @@
...
+                            data-confirm-title== t("linked_document.linked_document")]
+                            = t("linked_document.delete")
+
+                .spreadsheet-links.view-element
+                  - if document
+                    .doc-actions
+                      - if document.display_url.present?
+                        / Beware, the display_height value doesn't not seem to be honored
```
Comment (linesarefuzzy) [inline app/views/admin/loans/content/_questionnaire_group.html.slim]: Is this still true?

File: config/locales/en/loan_questions.en.yml [hunk 8]
```diff
@@ -17,6 +17,18 @@ en:
...
+  linked_document:
+    cell_end: End Cell
+    cell_end_hint: Z26
+    cell_start: Start Cell
+    cell_start_hint: A1
+    delete: Delete Link
+    delete_confirmation: Are you sure you want to remove this link?
+    link: "Link to file (such as a Google spreadsheet or document)"
+    linked_document: Linked Document
+    range_of_cells: Optional range of spreadsheet cells
```
Comment (linesarefuzzy) [inline config/locales/en/loan_questions.en.yml]: I prefer "Optional range for Google spreadsheet"—it will only work for Google spreadsheets anyway.

**Comment by linesarefuzzy** [comment]
Looks good. Just a few minor suggestions.

File: app/views/admin/loans/content/_questionnaire_group.html.slim (hunk 5 above)
Comment (emitche) [inline app/views/admin/loans/content/_questionnaire_group.html.slim]: Yes. I thought that looked odd. Will fix.

File: app/views/admin/loans/content/_questionnaire_group.html.slim (hunk 7 above)
Comment (emitche) [inline app/views/admin/loans/content/_questionnaire_group.html.slim]: It is. That was Joseph's note. And he had the display_height commented out and the box set to a height of about 500px.

The display_height only changes for content if both a starting cell and ending cell are defined. If not, the height is big. Even with those start and end cells, the height still isn't exactly the height of the content, but is a tad bit bigger. This only applies to spreadsheets (being able to adjust the height with start and end cell).

File: config/locales/en/loan_questions.en.yml (hunk 8 above)
Comment (emitche) [inline config/locales/en/loan_questions.en.yml]: Okay. I think Roxana made this change, but I will check.

File: app/views/admin/loans/content/_questionnaire_group.html.slim (hunk 6 above)
Comment (emitche) [inline app/views/admin/loans/content/_questionnaire_group.html.slim]: Very good to know that there are defaults. Thanks!

**Pull request merged**